roles/solrcloud/README.md

```diff
@@ -1,3 +1,138 @@
-## SolrCloud
+# SolrCloud Ansible Role
 
-This role will install [SolrCloud](https://cwiki.apache.org/confluence/display/solr/SolrCloud) on at least 3 endpoints listed in your inventory file.
+This Ansible role installs and configures Apache Solr in cloud mode (SolrCloud), supporting both Solr 8.x and Solr 9.x versions. It uses a direct file installation approach for better reliability and control.
+
+## Features
+
+* Direct file installation of SolrCloud (no reliance on install script)
+* Support for both Solr 8.x and Solr 9.x versions
+* ZooKeeper integration for cloud mode
+* Consistent directory structure and paths
+* Comprehensive logging configuration
+* Performance tuning with proper JVM settings
+* Modern systemd service management
+* Security configuration options
+
+## Requirements
+
+* Ansible 2.9 or higher
+* Debian/Ubuntu-based target systems
+* Java 17+ installed on target servers
+* ZooKeeper ensemble configured and running
+
+## Role Variables
+
+All variables are defined in `defaults/main.yml` with sensible defaults. Here are the key variables you may want to customize:
+
+### Version Settings
+
+```
+yaml
+solr_cloud_download_version: "9.2.0"  # Solr version to install
+solr_cloud_url: "https://mirror-server/solr/{{ solr_cloud_download_version }}/{{ solr_cloud_package }}"
+```
+
+### User and Directories
+
+```
+yaml
+solr_user: deploy
+solr_group: deploy
+
+solr_home: /solr
+solr_data_dir: "{{ solr_home }}/data"
+solr_log_dir: "{{ solr_home }}/logs"
+
+solr_installation: "/opt/solr"  # Symlink to versioned directory
+solr_versioned_dir: "/opt/solr-{{ solr_cloud_download_version }}"
+```
+
+### JVM Settings
+
+```
+yaml
+solr_heap: "20g"  # JVM heap size
+java_home: "/usr/lib/jvm/java-17-openjdk-amd64"
+```
+
+### ZooKeeper Configuration
+
+```
+yaml
+solr_zookeeper_hosts:
```

Comment:
This is the third var with very similar content - `zookeeper_hosts`, `zk_nodes`, and now `solr_zookeeper_hosts`. Can we consolidate?